Expanded contributors guide and updated to match latest website layout norms (#237)
risc0/website: docs/contributors-guide.md

* [Make website suggestions or report bugs](https://github.com/risc0/website/issues) -- Submit suggestions and issues via GitHub issues
* [Create a new page](https://www.github.com/risc0/website) - Use the GitHub browser interface to create new content in `src/pages` and `docs`
  * Please read our documentation best practices and navbar and sidebar modification instructions below before creating new pages.
* [Give feedback on open pull requests](https://github.com/risc0/website/pulls) -- We welcome community feedback on changes
* [Contribute to the main zkVM project](https://github.com/risc0/risc0) -- Submit revisions and make change suggestions
* [Ask questions on Discord](https://discord.gg/risczero)

You might need these setup steps:
* [Set up a GitHub account](https://github.com/) -- If you don't have an account, you'll need one for most contributions here
* [Clone the repository and work locally](https://docusaurus.io/docs/installation) - Check the Docusaurus installation instructions.
  * To run a local build, we like `yarn start`, especially when we're working with changes that involve links or sidebars. 

> # Website contribution style guidelines

Reviewer: Also should not be a top level heading
```suggestion
## Website contribution style guidelines
```

# Contributor's Guide
*This page describes guidelines for community contributions. RISC Zero welcomes community participation!*

Broadly speaking, we have three ways to contribute as a community member:

> 1. You can make suggestions or report bugs via GitHub issues for the [main zkVM project](https://github.com/risc0/website/issues) and the [website](https://github.com/risc0/website/issues). 

Reviewer 2: I'd love to either start these with a short **boldface** tag or make them more succinct -- perhaps both. 
Compare to ethereum's contributing guide: https://ethereum.org/en/contributing/

Reviewer 2: I think the prior version is closer to what I'd like to see at the top of the page here

<img width="471" alt="image" src="https://github.com/risc0/website/assets/15272444/8530b6d2-cb38-40b6-a9a8-e1de4917d714">

Author: Created bold tags, but tacked on action-specific instructions so they'd be colocated.

Reviewer 2: This section still feels like way too much text to me. 

Compared to the current style, I highly prefer both the prior version and the approach used on this page: https://ethereum.org/en/contributing/

2. We welcome proposed changes in the form of pull requests or comments on outstanding pull requests against the [main zkVM project](https://github.com/risc0/risc0/pulls) and [website](https://github.com/risc0/website/pulls).
3. Contribute to the [community Discord](https://discord.gg/risczero). At this time we are not looking for moderators, but we love questions and posts about performance benchmarks!

# Contribution highlights

To get you started cooking up contribution ideas, we've highlighted a few interesting ways to participate in our community.

> ### Add tutorials and how-to guides to the website

Reviewer 2: Do we want to be encouraging this? What sort of tutorials or how-to guides would we hope to see?

Author: This is a move of 
">- Contribute to our tutorials and how-to guides for our [templates](https://github.com/risc0/risc0/tree/main/templates) an
d [Rust examples](https://github.com/risc0/risc0/tree/v0.16.0/examples)"
If we want to suggest editing the existing guides (or remove the suggestion altogether) that's fine too.

Reviewer 2: I think our documentation has matured to the point that if we're asking people to write tutorials or how-to guides, we should have an itemized list of guides we're looking for

Reviewer: I also think we don't particularly want to be encouraging external tutorials/how-tos. We'd certainly accept one if a well-written PR came in, but I think we're going to get a bunch of low effort confused examples if we include this here.

Additionally: I could be talked into including this here eventually, but at the moment we should not have it because we're about to be making substantial example revisions (@pdg744 to the readmes and me to the code organization in the starter template and correspondingly in the examples). We shouldn't encourage people to start their own examples when we're about to change things out from under them.

So, in this PR, drop this section. If we want it back later, let's open a separate PR for that.

We welcome additional tutorials and how-to guides for our [templates](https://github.com/risc0/risc0/tree/main/templates) and [Rust examples](https://github.com/risc0/risc0/tree/main/examples).

### Suggest terminology and naming convention improvements
[`RISC Zero Official Terminology`](./terminology)

Our terminology and naming conventions are subject to ongoing evaluation, and we encourage conversation and questions on these topics. Please let us know via a [`GitHub issue`](https://github.com/risc0/website/issues) when you encounter terms that don't seem quite right.

> ### Contribute benchmarks for the main zkVM project

Reviewer: IMO this subsection should be the first one listed (but I'm open to being overruled).

> We'd love to see your benchmarks! You can add these to the `suggest-server-setup` channel on our [community Discord](https://discord.gg/risczero) (or better yet, write a blog post and link to it in the Discord!) 

Reviewer 2: `suggest-server-setup` does not seem like the appropriate channel to me -- that channel is intended for suggestions about how to set up the discord server

Author: Moving to general for now until we can talk about other channels. The community contribution options are under active consideration so I'll leave off on anything dedicated for now.

Author: Updating this channel suggestion to `general` for now.

Reviewer 2: I don't think this comment has been addressed yet

> Here's how you benchmark:

Reviewer 2: I don't think we want to put instructions for how to benchmark on this page -- from a maintenance perspective, I think we're better off putting this info in the `Performance` section of the zkVM docs.

Author: These additions are coming from decisions about a minimum set of expanded suggestions for the community contribution guide and I do want to include reference to benchmarking here, but I can document the full set of instructions in `Performance` and link to it.

Reviewer 2: mentioning benchmarking here sounds great, and adding documentation to the website explaining how to do benchmarking sounds fantastic

Reviewer 2: looks like there's already instructions here: https://dev.risczero.com/zkvm/benchmarks#generating-your-own-benchmarks

Reviewer 2: I don't think this comment has been addressed yet

> 1. Pull down the [main zkVM repository]((https://github.com/risc0/risc0).

Reviewer: Broken link (but maybe this will be dropped after switch to the benchmark reference @pdg744 mentioned upthread?)
```suggestion
1. Pull down the [main zkVM repository](https://github.com/risc0/risc0).
```

We also keep content digestible by splitting long document sections into smaller chunks.
>- `Progressive Disclosure`: Our landing pages are simple and clear. 
Both at the level of site-organization and individual doc-organization, we present a bird's eye view first with opt-in paths toward higher levels of detail and technicality. 
>- `Discoverability`: Anyone who finds their way to any RISC Zero content is able to rapidly find their way to the material that best suits their needs.
When contributing new content, we consider document placement and ease of navigation to and from related materials.
>- `Modular Components`: We keep materials succinct through extensive use of modular, single-purpose documentation pages.
Where possible, we split documentation into single-purpose pages and rely on navigation links as pointers to additional information.
>- `Consistent and Accessible Terminology`: We are diligent about using our official terminology as defined and using precise language as much as possible. 
At the same time, we minimize our use of technical jargon, taking care to provide reference pages to prerequisite knowledge as appropriate. 

> ### Editing the navbar and sidebar 

Reviewer 2: (seems fine to me to remove this section altogether; seems very granular for a contributor's guide)

> ### Editing the navbar and sidebar 

Reviewer 2: I don't think this comment has been addressed yet

*This page describes guidelines for community contributions. RISC Zero welcomes community participation!*

## How To Contribute
- All changes to this website are managed through GitHub pull requests, so you'll need a [GitHub Account](https://github.com/) to contribute. 
- You can suggest an edit directly via the `Edit this Page` button at the bottom of each page. 
- To create a new page, you can use the [GitHub browser interface](https://www.github.com/risc0/website); the content is in `src/pages` and `docs`. 

>   - Please read about [the navbar and sidebars](./contributors-guide.md/#navbar-and-sidebars) and [categories of documentation](./contributors-guide.md/#categories-of-documentation) before creating a new page.

Reviewer: This PR just dropped the navbar & sidebars section ... either drop this part or reduce it to a mention of ensuring you add the page to the sidebar
```suggestion
  - If you're creating a new page, please read about [categories of documentation](./contributors-guide.md/#categories-of-documentation) and add your page to the appropriate section of the sidebar.
```
